### PR TITLE
Use PascalCase for enum values

### PR DESCRIPTION
**Pull request by ChongTang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ChongTang** [comment]
Hi @heaths , I renamed LUIS to Luis to align with our authoring Swagger.

File: specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json
Comment (tjprescott) [inline specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Are you intentionally keeping it all caps in the docs?

**Review by tjprescott — COMMENTED** [review]
The breaking change check passed but I'm curious how this would not result in breaking changes to generated code as it would change the name of the enum label.

File: specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Yes. LUIS in the doc is intended because that is the name of our GA product.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: I am not sure if this will break code generation. I think No because CLU Authoring made the same change. @heaths ?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: It will - most languages are case-sensitive - but you're breaking a preview. Being consistent across COGS Language pillar before we GA is worth it, IMO.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: I agree it's worth it. This is our final decision to be consistent across at least CLU authoring and runtime, as Abbas is correcting these in Authoring Swagger. Do I need to update anything in this PR, like renaming `LUISParameters` to `LuisParameters`?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: FWIW, this seems to be based on the guidelines saying acronyms should be PascalCase, but I believe that was only intended for models and property names. I had a comment in another PR and tagged a few people for follow-up. If you have the time and would prefer "LUIS", we might want to wait for an answer.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Sounds good. I will wait for it. The back-end service needs more time to change, but this renaming to the class names does not require back-end changes. So we should be good.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Hi @heaths , is there any update on this? I can update LUIS to Luis. I am totally OK with that.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: For model names and property names, this should be `Luis`. I was referring to property values, which you also changed. @markweitzel @mikekistler do I have that correct - only model/property names should use "Luis" while enum values, for example, can use "LUIS", or should they all use "Luis". Specs aren't crystal clear.

Comment (mikekistler) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: We have not included a convention for enum values in the guidelines _yet_, but the issue has come up previously and after discussion the consensus was that enum values should also be camel case.  I captured this in this issue:

https://github.com/microsoft/api-guidelines/issues/342

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: I'll comment on that, but I would assume that acronyms would be all lowercase then e.g., "luis".

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: So are we going to use all lowercase for all enum values?

Comment (mikekistler) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: The proposal -- not yet codified in the API Guidelines -- is that enum values should be camel case (specifically _lower_ camel case -- i.e. the first letter is always lower case). And acronyms should be treated as "words" in this context, so in this particular case the recommendation would be "luisParameters".

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: @heaths If we change the model name as `luisParameters`, will there be any code generation issue in some languages?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: No. The languages' code generators will use the proper naming conventions e.g., Python will generate `luis_parameters` and C# will generate `LuisParameters`.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Sounds good. I will update the model name to `luisParameters`.

For enum values, we will use `luis`, `orchestration`, `conversation`, and `questionAnswering`, `nonLinked`. Does this look good to you?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Yes. I would also update the title of this PR since it will be saved to history forever. Might as well make the title right for easier troubleshooting.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Hi @moabba, after some discussion, the board recommended to use camel case for enum values. I am sorry for the back and forth. Can you update the authoring to use the same values?

For enum values, we will use `luis`, `orchestration`, `conversation`, `questionAnswering`, `nonLinked`.

Comment (moabba) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: The authoring uses camelCase for all the enum values. We had to change the project type casing to PascalCase to match both CLU and CT prediction.

Text Analytics use a lot of different variants, but the task kinds (project types in authoring) are PascalCase. I would recommend keeping this convention for consistency.

If we had enough extra time, I would've asked TA to use camelCase, and then use camelCase for the enum values you're mentioning. But unfortunately, we don't have this time (GA is in 2 weeks).

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: @heaths , I totally agree with Abbas. Both TA and CLU runtime are using PascalCase right now, it would require all services to update in a short time before GA. Are we OK with PascalCase right now?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: That's fine, but I would make it consistent between your inference and authoring swaggers.

Remember, the guidelines as currently written were only about model and property names. If there's no time to change those I suppose that's fine - our SDKs should still do the right thing (we can fix them if not); however, the property values weren't necessary to change as currently documented. Mike only mentioned that we were going to update the guidelines to recommend camelCase.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: After this PR get merged, the inference runtime and authoring will be consistent. I guess that means there is no more update to this PR. Heath, if you can approve it then we can merge it.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: I'm confused. I thought the models you renamed to `LuisParameters` would need to be reverted to `LUISParameters` then. Or are they already? This has gone back and forth enough I have to admit I'm confused what the desired casing is for:

* Models
* Model properties
* Property values e.g., enum values

Could you clarify?

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: Sorry for the confusion. `LUISParameters` is unique to the runtime. This PR means to align enum values with the authoring side. 

The change on model name `LUISParameters` is to simply unify what we used in enum values. Do you suggest I revert the model name back and only change enum values in this PR?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json:315]: The recommendations are (or will be):

* Models are PascalCase
* Model properties are camelCase
* Property / enum values are camelCase

I would try to align to those as much as you can / are comfortable with now. But please let me know in a comment. I would like to review all the swaggers for GA to see if we're consistent, or is consistency not achievable in the necessary timeframe?

**Review by heaths — APPROVED** [review]

**Pull request merged**
